contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -1,32 +1,61 @@
+use std::cmp::Ordering;
+
 use cosmwasm_std::{
-    to_binary, Api, Binary, CanonicalAddr, CosmosMsg, Empty, Env, Extern, HandleResponse,
+    log, to_binary, Api, Binary, CanonicalAddr, CosmosMsg, Empty, Env, Extern, HandleResponse,
     HumanAddr, InitResponse, Order, Querier, StdError, StdResult, Storage,
 };
+
+use cw0::Expiration;
 use cw2::set_contract_version;
+use cw3::{
+    ProposalListResponse, ProposalResponse, Status, ThresholdResponse, Vote, VoteInfo,
+    VoteListResponse, VoteResponse, VoterListResponse, VoterResponse,
+};
 
 use crate::msg::{HandleMsg, InitMsg, QueryMsg};
 use crate::state::{
     ballots, ballots_read, config, config_read, next_id, parse_id, proposal, proposal_read, voters,
     voters_read, Ballot, Config, Proposal,
 };
-use cw0::Expiration;
-use cw3::{
-    ProposalListResponse, ProposalResponse, Status, ThresholdResponse, Vote, VoteInfo,
-    VoteListResponse, VoteResponse, VoterListResponse, VoterResponse,
-};
-use std::cmp::Ordering;
 
 // version info for migration info
 const CONTRACT_NAME: &str = "crates.io:cw3-fixed-multisig";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 pub fn init<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
-    _env: Env,
+    env: Env,
     msg: InitMsg,
 ) -> StdResult<InitResponse> {
+    if msg.required_weight == 0 {
+        return Err(StdError::generic_err("Required weight cannot be zero"));
+    }
+    if msg.voters.is_empty() {
+        return Err(StdError::generic_err("No voters"));
+    }
+    let weights: StdResult<Vec<u64>> = msg
+        .voters
+        .iter()
+        .map(|v| {
+            if v.weight == 0 && v.addr != env.message.sender {
```

Comment:
After your argument why 0 could make sense, I am happy for any of these to be 0, as long as the other checks (above and below) pass.